GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -373,6 +375,75 @@ void ManageWorktreeSupport()
             }
         }
 
+        private readonly Regex _aheadBehindRegEx = new Regex(@"^\[(ahead (?<ahead>\d+))?(, )?(behind (?<behind>\d+))?\] (?<branch>.*)$", RegexOptions.Compiled | RegexOptions.Multiline);
+
+        private readonly TranslationString _aheadCommitsToPush = new TranslationString("{0} new commit(s) will be pushed");
+        private readonly TranslationString _behindCommitsTointegrateOrForcePush = new TranslationString("{0} commit(s) should be integrated (or will be lost if force pushed)");
+
+        private const string aheadBehindGitCommand = "for-each-ref --format=\"%(push:track) %(refname:short)\" refs/heads/";
+        private void DisplayAheadBehindInformation()
+        {
+            if (!GitVersion.Current.SupportAheadBehindData)
+            {
+                return;
+            }
+
+            toolStripButtonPush.DisplayStyle = ToolStripItemDisplayStyle.Image;
```

Comment:
I am not content with adding more code to `FormBrowse`, it is already several thousand lines long.
This functionality seems fairly self contained and could be abstracted in a form of a usercontrol/custom toolstrip button.